### ISSUE TITLE
feat: external managed cluster registration and node handling

### ISSUE THREAD

**Issue by chrxmvtik** [apache/cloudstack-kubernetes-provider#73]
Hello fellas,

Since cloudstack kubernetes provider is successor of cloudstack cloud controller, shouldn't it also allow to register external managed clusters automatically? I am not saying it should be mandatory, but would be great if we had a choice. So it would fetch information about nodes and register cluster to the cloudstack, besides that it could handle node removal if virtual machine is removed and so on.

**Comment by Pearl1594** [comment]
Currently, CAPC clusters - K8S clusters deployed using cluster API on CloudStack, can be viewed on CloudStack UI; with the UI providing a centralized view to all of the ACS resources used by this external cluster.
In the upcoming release of CloudStack, we have added support to import external nodes - VMs or baremetal/physical servers as worker nodes to managed CKS clusters.

**Comment by chrxmvtik** [comment]
> Currently, CAPC clusters - K8S clusters deployed using cluster API on CloudStack, can be viewed on CloudStack UI; with the UI providing a centralized view to all of the ACS resources used by this external cluster. In the upcoming release of CloudStack, we have added support to import external nodes - VMs or baremetal/physical servers as worker nodes to managed CKS clusters.

Yes, that is one way to achieve that, however we don't always have possibility to deploy using CAPI and we have to stick to what we have. Hence why I opened this issue 🚀
